packages/pxweb2-ui/src/lib/components/BottomSheet/BottomSheet.tsx

```diff
@@ -0,0 +1,105 @@
+import cl from 'clsx';
+import { useEffect, useRef, useState } from 'react';
+
+import classes from './BottomSheet.module.scss';
+import Heading from '../Typography/Heading/Heading';
+import Button from '../Button/Button';
+
+export interface BottomSheetProps {
+  readonly heading: string;
+  readonly closeLabel?: string;
+  readonly isOpen: boolean;
+  readonly onClose?: () => void;
+  readonly className?: string;
+  readonly children: React.ReactNode;
+}
+
+export function BottomSheet({
+  heading,
+  closeLabel = 'Close',
+  isOpen,
+  onClose,
+  className = '',
+  children,
+}: BottomSheetProps) {
+  const cssClasses = className.length > 0 ? ' ' + className : '';
+  const [isBottomSheetOpen, setIsBottomSheetOpen] = useState(isOpen);
+  const bottomSheetRef = useRef<HTMLDialogElement | null>(null);
+  const [isClosing, setIsClosing] = useState(false);
+
+  useEffect(() => {
+    setIsBottomSheetOpen(isOpen);
+  }, [isOpen]);
+
+  useEffect(() => {
+    const modalElement = bottomSheetRef.current;
+    if (modalElement) {
```

Comment:
Consider negating this and returning, instead of nesting the code within


If(!modalElement) {
  return
}